usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -0,0 +1,227 @@
+                cursor.execute(clear_table_sql)
+            logger.info(f"{temp_destination_table} dropped.")
+
+        # Recreate the table if it doesn't exist. Spark's df.write automatically does this but doesn't account for
+        # the extra stuff (indexes, constraints, defaults) which CREATE TABLE X LIKE Y accounts for.
+        # If there is no source_table to base it on, it just relies on spark to make it and work with delta table
+        if source_table or source_schema:
+            if source_table:
+                create_temp_sql = f"""
+                    CREATE TABLE {temp_destination_table} (
+                        LIKE {source_table} INCLUDING DEFAULTS INCLUDING IDENTITY
+                    )
+                """
+            else:
+                create_temp_sql = f"""
+                    CREATE TABLE {temp_destination_table} (
+                        {", ".join([f'{key} {val}' for key, val in source_schema.items()])}
+                    )
+                """
+            with db.connection.cursor() as cursor:
+                logger.info(f"Creating {temp_destination_table}")
+                cursor.execute(create_temp_sql)
+                logger.info(f"{temp_destination_table} created.")
+
+                # Copy over the constraints before indexes
+                # Note: we could of included indexes above (`INCLUDING CONSTRAINTS`) but we need to drop the
+                #       foreign key ones
+                if source_table:
+                    logger.info(f"Copying constraints over from {source_table}")
+                    copy_constraint_sql = make_copy_constraints(
+                        cursor, source_table, temp_destination_table, drop_foreign_keys=True
+                    )
+                    if copy_constraint_sql:
+                        cursor.execute("; ".join(copy_constraint_sql))
+                    logger.info(f"Constraints from {source_table} copied over")
+
+            # Converting unsupported data types on spark to their delta equivalent
+            # We will cast these back to their original intended types afterwards
+            # Only applicable if we're copying down a table that was already in postgres
+            alter_column_sql = []
+            source_columns = []
+            if source_model:
+                source_columns = [(column.name, type(column)) for column in source_model._meta.get_fields()]
+            else:
+                source_columns = list(source_schema.items())
+
+            for column_name, column_type in source_columns:
+                if column_type in SPECIAL_TYPES_MAPPING:
+                    special_columns[column_name] = column_type
+                    delta_type = SPECIAL_TYPES_MAPPING[column_type]["delta"].format(column_name=column_name)
+                    alter_column_sql.append(
+                        f"ALTER TABLE {temp_destination_table}" f" ALTER COLUMN {column_name} TYPE {delta_type}"
```

Comment:
Will the table be empty when making this alteration? If not, it could take a long long time to execute.

Reply:
Table should be empty at this point. If it already existed, we've cleared it out and rebuilt it without any data.